lib/fs.js

```diff
@@ -340,6 +340,10 @@ fs.accessSync = function(path, mode) {
 fs.exists = function(path, callback) {
   if (handleError((path = getPathFromURL(path)), cb))
     return;
+  if (typeof path !== 'string' && !(path instanceof Buffer)) {
```

Comment:
Should these type check be placed at the beginning of these methods?

Reply:
@starkwang No because the check must happen after `getPathFromURL(path)` is called. The `path` may be a string, a Buffer, or a URL object.